go/types/blob.go

```diff
@@ -34,6 +34,27 @@ func (b Blob) Reader() io.ReadSeeker {
 	return &BlobReader{b.seq, cursor, nil, 0}
 }
 
+func (b Blob) Splice(idx uint64, deleteCount uint64, data []byte) Blob {
+	if deleteCount == 0 && len(data) == 0 {
+		return b
+	}
+
+	d.Chk.True(idx <= b.Len())
+	d.Chk.True(idx+deleteCount <= b.Len())
+
+	cur := newCursorAtIndex(b.seq, idx)
```

Comment:
2 main comments:
1. A more Go-ish interface here would be to implement Splice using the Go `io.*` libraries. I can't think of a good way to do it though :\
2. We should file an issue to make this look like NewStreamingBlob. Boxing/unboxing each byte is super inefficient, which is why that custom chunking code exists. If Splice becomes performance sensitive, especially if `data` becomes large, then we should fix it.